client/my-sites/site-settings/seo-settings/form.jsx

```diff
@@ -512,4 +485,14 @@ const mapDispatchToProps = {
 		recordTracksEvent( 'calypso_seo_tools_front_page_meta_updated', properties ),
 };
 
-export default connect( mapStateToProps, mapDispatchToProps )( protectForm( localize( SeoForm ) ) );
+const withCurrentSiteKey = createHigherOrderComponent(
```

Comment:
Instead of having to create a HoC just to key by site ID, can we just pass the site ID [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/marketing/traffic/index.js#L51) as a prop?

Reply:
I don't know. Wouldn't that mean we don't just get a new `SiteSettingsSEOForm` but a new `AsyncLoad` wrapper component as well? I've gotta try.

Reply:
I think we should be good; we shouldn't get a new `AsyncLoad` wrapper component. `AsyncLoad` will load the component only once, and will pass the new props down to it if they get updated at some point.

Reply:
Fixed via https://github.com/Automattic/wp-calypso/pull/59940/commits/cc01884b2e95514ec6aafb123f8bc7e14f57d9fc – I don't think `key` is passed down but it's still working as expected and removes the need for an HOC.